### ISSUE TITLE
Some hidden content still save in *.html

### ISSUE THREAD

**Issue by issuex** [gildas-lormeau/SingleFile#1110]
**Expected behavior**
SingleFile should not download the hidden content.

**Screenshots**
<img width="586" alt="report-them" src="https://user-images.githubusercontent.com/111990330/216370425-76caa8ac-f392-4271-a7ea-cf2f99214b9a.png">

**Environment**
 - OS: Win11 Pro
 - Browser: Firefox
 - Version: 64

**Additional context**
Seem like the page hide there content by remove `in active` 
From my test add `in active` make the content show up, remove it the content was gone.
Not sure is it possible to detect this kind of content.

**Comment by gildas-lormeau** [comment]
If it is not removed from the DOM then it means that SingleFile is not able to detect the content is actually hidden. SingleFile takes some precautions to make sure DOM deletions don't break the document. This is probably the cause of the issue.

**Comment by GlassGruber** [comment]
Sorry to bump this, is this intended also when option `remove hidden elements` is checked in _HTML Content_? 

Just asking because I tried to save a portion highlighted of a page that contained some images and a `<video>` and the resulting html file contained all elements even if not selected.

An example of page can be found here 
https://yoast.com/google-discover/

Thank you for your time!

**Comment by gildas-lormeau** [comment]
@GlassGruber Can you please give me a procedure to reproduce the issue? I did a test but I was not able to reproduce it with the webpage you mentioned.

**Comment by GlassGruber** [comment]
> @GlassGruber Can you please give me a procedure to reproduce the issue? I did a test but I was not able to reproduce it with the webpage you mentioned.

I've selected a portion of the page in the middle of the page with text only and then 
right click > Save selection. 
I should have mention that I'm using firefox with last Version 1.21.42. 

Also I want to make it clear, the selected content is the only thing visible as expected. 
What I'm referring to is that the rest of the page, including a video and couple images, are part of the _source code_ of the resulting HTML file, with their src encoded as BASE64. 

I noticed this only because I was surprised that a couple paragraph snippet weighed about 4+ MB! 

Thank you for looking at this! Hopefully is not an RTFM issue 🙃
